Terraform/AWS/VPCinAWS.tf

```diff
@@ -0,0 +1,135 @@
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+  tags = {
+  Name = "VPC_Main"
+  }
+}
+resource "aws_subnet" "public1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "public1"
+  }
+}
+resource "aws_subnet" "public2" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.2.0/24"
+
+  tags = {
+    Name = "public2"
+  }
+}
+resource "aws_subnet" "private1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.3.0/24"
+
+  tags = {
+    Name = "private1"
+  }
+}
+resource "aws_subnet" "private2" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.4.0/24"
+
+  tags = {
+    Name = "private2"
+  }
+}
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "IGW"
+  }
+}
+resource "aws_nat_gateway" "natgw1" {
+  subnet_id     = aws_subnet.private1.id
+  allocation_id = aws_eip.eip1.id
+  tags = {
+    Name = "natgw1"
+  }
+
+  # To ensure proper ordering, it is recommended to add an explicit dependency
+  # on the Internet Gateway for the VPC.
+  depends_on = [aws_internet_gateway.igw]
+}
+resource "aws_nat_gateway" "natgw2" {
+  subnet_id     = aws_subnet.private2.id
+  allocation_id = aws_eip.eip2.id
+  tags = {
+    Name = "natgw2"
+  }
+}
+resource "aws_route_table" "public1" {
+  vpc_id = aws_vpc.main.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.igw.id
+  }
+
+  tags = {
+    Name = "public RT1"
+  }
+}
+resource "aws_route_table" "public2" {
```

Comment:
It is okay to use one public route table for the two public route table association as they are referencing to the same internet gateway but for the private route table you are to use 2 route table as they will reference the two nat gateway each respectively

Reply:
@geoffrey1330 Thank you, please review the new commit, I removed one public RT and associate the second RT association to the same RT